datafusion/sql/tests/sql_integration.rs

```diff
@@ -4530,79 +4552,72 @@ fn roundtrip_expr() {
-            "SELECT p1.id, COUNT(*) AS cnt FROM (SELECT p1.id FROM person AS p1 JOIN person AS p2 ON (p1.id = p2.id)) GROUP BY p1.id"
-        ),
-        (
             "select id, count(*), first_name from person group by first_name, id",
-            "SELECT person.id, COUNT(*), person.first_name FROM person GROUP BY person.first_name, person.id"
-        ),
-    ];
+            "select id, sum(age), first_name from person group by first_name, id",
+            "select id, count(*), first_name 
+            from person 
+            where id!=3 and first_name=='test'
+            group by first_name, id 
+            having count(*)>5 and count(*)<10
+            order by count(*)",
+            r#"select id, count("First Name") as count_first_name, "Last Name" 
+            from person_quoted_cols
+            where id!=3 and "First Name"=='test'
+            group by "Last Name", id 
+            having count_first_name>5 and count_first_name<10
+            order by count_first_name, "Last Name""#,
+            r#"select p.id, count("First Name") as count_first_name,
+            "Last Name", sum(qp.id/p.id - (select sum(id) from person_quoted_cols) ) / (select count(*) from person) 
+            from (select id, "First Name", "Last Name" from person_quoted_cols) qp
+            inner join (select * from person) p
+            on p.id = qp.id
+            where p.id!=3 and "First Name"=='test' and qp.id in 
+            (select id from (select id, count(*) from person group by id having count(*) > 0))
+            group by "Last Name", p.id 
+            having count_first_name>5 and count_first_name<10
+            order by count_first_name, "Last Name""#,
+        ];
 
-    let roundtrip = |sql: &str| -> Result<String> {
+    for query in tests {
         let dialect = GenericDialect {};
-        let statement = Parser::new(&dialect).try_with_sql(sql)?.parse_statement()?;
+        let statement = Parser::new(&dialect)
+            .try_with_sql(query)?
+            .parse_statement()?;
 
         let context = MockContextProvider::default();
         let sql_to_rel = SqlToRel::new(&context);
-        let plan = sql_to_rel.sql_statement_to_plan(statement)?;
+        let plan = sql_to_rel.sql_statement_to_plan(statement).unwrap();
 
-        println!("{}", plan.display_indent());
+        let roundtrip_statement = plan_to_sql(&plan)?;
 
-        let ast = plan_to_sql(&plan)?;
+        let actual = format!("{}", &roundtrip_statement);
+        println!("roundtrip sql: {actual}");
+        println!("plan {}", plan.display_indent());
 
-        println!("{ast}");
+        let plan_roundtrip = sql_to_rel
+            .sql_statement_to_plan(roundtrip_statement.clone())
+            .unwrap();
 
-        Ok(format!("{}", ast))
-    };
-
-    for (query, expected) in tests {
-        let actual = roundtrip(query).unwrap();
-        assert_eq!(actual, expected);
+        assert_eq!(plan, plan_roundtrip);
```

Comment:
As mentioned in the description, the key condition tested is that the logical plan derived from the original AST and recovered AST are the same logical plan.

Reply:
I think it makes sense

Can you please add a comment to this test that explains the rationale (that you so eloquently explained in the PR' description)? I think it will help future readers understand the choice which may not be obvious

Reply:
:heavy_check_mark: